### DISCUSSION TITLE
조직 내 커뮤니케이션(디자이너, 기획자) 측면에서 문제점 파악

### DISCUSSION THREAD

**Discussion by peanudge** [InnerCircleA/design-system#10] [Show and tell]
각자의 조직내 커뮤니케이션(디자이너, 기획자) 측면에서 문제점을 파악해서 공유해주세요!

각자 보고서 양식으로 공유해도 좋겠지만 서로 관련된 의견을 주고 받는게 더 의미있고 discussion에 남겨야 더 오래 볼 것같아서 여기 기록해두면 좋을 것 같아요.

**Comment by peanudge** [comment]
# 🧑‍💻 디자이너와 기획자와 커뮤니케이션 문제점 파악

현재 IT 서비스를 만드는 스타트업 회사에서 핵심 인력은 아무래도 product를 만드는 기획자 + 디자이너 + 개발자 입니다. 

빠르게 product를 만들어서 배포하고 고객의 피드백을 받은 후 데이터 기반으로 더 나은 방향으로 product를 개선하는 것이 저희 팀의 목표입니다.

이렇듯 속도가 생명인 상황에서 커뮤니케이션 이슈는 현재 조직의 큰 문제이지만 크게 문제 삼는 것이 조심스러운 상황입니다. 

혹은 크게 문제삼아도 **엄격한 문서화** 혹은 강한 피드백이 해결책으로 사용될 경우 오히려 커뮤니케이션을 단절시키는 경향이 있었습니다. 

모두가 커뮤니케이션 안하는 것보다는 **자유롭게 하는 것이 더 낫다고 판단**해서 불필요한 커뮤니케이션을 어쩔 수 없이 받아들이는 현상태입니다. 

현재 상태를 간략히 남겼고 기획자, 디자이너와의 커뮤니케이션을 구체화해서 정리해보겠습니다.

# 디자이너와 커뮤니케이션에서의 문제 

**개발자 수준에서의 디자인 시스템 부재**로 잦은 디자인 커뮤니케이션을 자주하게 되는 상황이 연출됩니다.

ex) 개발자가 다루는 레이아웃 시스템(`flex`,`grid`)과 디자이너가 구성하는 레이아웃이 다름, foundation 일관성 QA를 매번 해야함.

해결책으로 디자인 시스템을 만들면 되지않을까 싶지만 **디자인 시스템을 만들 경험과 시간**이 부족하기에 선뜻 진행하기 어렵습니다.

그리고 업무 효율 개선과 기능 개선을 비교했을 때 **업무 효율 개선은 기능개선에 항상 우선순위**에 밀리게 됩니다.

- 디자인 시스템을 대체할 만큼 잦은 커뮤니케이션 발생
- 디자인 시스템을 만들 경험, 리소스 부족

결국, 잦은 디자인 관련 커뮤니케이션을 지속적으로 진행해야하는 상황입니다.

> 프러덕트 디자이너들의 역할은 기획자들과 맞물려있어서 추가적인 문제는 아래 기획자와의 커뮤니케이션에서 이어서 다루겠습니다.

# 기획자와의 커뮤니케이션 문제

기획자는 서비스가 잘되게 하기위해 모든 일을 하며 특히 데이터 기반으로 Product 기능을 추가 & 개선 작업을 기획하게 됩니다.

기획자 업무의 산출물은 PRD(Product Requirement Document, 기획서)가 되며 **관련 담당자들과 커뮤니케이션이 모두 이루어지고** 완성되게 됩니다. 

여기서 발견한 첫번째 문제는 PRD가 나오기까지 커뮤니케이션이 충분히 이뤄지지 않는 경우가 대다수이며 디자인 도중, 개발 도중, 배포 후 계속해서 요구사항이 달라질 경우가 자주 있다는 것입니다. 빠르게 성장해야하고 만들어보고 피드백을 받는 문제가 강한 스타트업에서는 불가피한 일이지만 너무 잦으면 큰 문제가 발생할 가능성이 있습니다.

조직 내 기획자들과의 커뮤니케이션 문제 중 하나는 **데이터에 대한 이해가 다르다**는 것 입니다. 기획자들이 DB 테이블 구조, User 행동 데이터(GA, Amplitude)에 완벽한 이해를 하고 있으면 좋겠지만 일반적으로 상당한 경험이 있어야하면 숙련될 때까지는 많은 도움(커뮤니케이션)이 필요합니다.

- 조직 내 기획 직군은 리텐션이 낮아서 자주 교체되는 상황
- 모두가 DB 쿼리, 유저 행동 데이터 분석에 숙련된 것은 아님

그리고 또 다른 문제점 중 하나는 다양한 작업 요소들에 있어서 엔지니어와 차이가 크다는 것 입니다.

대표적인 작업 요소들은 **작업량**(size), **작업 기간**(time)이 있습니다.

- 작업량 예측이 어려움

새로운 백엔드 API 개발이 필요할지, 새로운 DB 테이블이 필요할지 등등 백엔드 개발자와의 커뮤니케이션을 잘해야지 일의 범위를 알 수 있습니다.

- 작업 기간 예측이 어려움

주니어 개발자와 시니어 개발자가 동일한 일을 했을 때 시간 차이를 알 수 없으며 이건 개발자들 간에도 쉽게 판단하기가 어려움.

사실 위 요소는 저희 조직만의 문제는 아니지만 커뮤니케이션을 충분히한다면 어느정도 서로 예측은 가능할 것입니다. 하지만 이런 커뮤니케이션의 필요성을 조직내에서는 새로운 기획자가 올때마다 해야하는 것에 피로를 느끼고 있습니다.

**Comment by Jeewhan** [comment]
# 커뮤니케이션(제품책임자, 디자이너) 측면에서 문제점 파악 후 문서화

## 기획과 관련한 문제점 파악

앞선 [개인 미션](https://github.com/InnerCircleA/design-system/discussions/9#discussioncomment-4451291) 에서 언급한 것처럼 제품의 방향성이 굉장히 자주 변경됩니다.

그럼에도 팀의 제품책임자 분이 대단히 뛰어나셔서 흔히 말하는 제품 관리 공백(Product Management Vacuum)에 해당하는 제품 비전, 제품 전략, 릴리스 계획 에 대한 공유가 데일리 미팅에서 매일 매일 굉장히 잘 이루어지고 있습니다.

또한 팀의 테크 리드가 일종의 스크럼 마스터 역할을 매우 훌륭히 해내고 있어서 무수한 피봇에도 불구하고 프로젝트가 잘 진행되고 있습니다.

하지만 문서화를 선호하지 않는 팀 분위기로 인해, 문서를 전혀 작성하지 않아 이후에 팀에 새로운 합류자가 발생한다면 제품의 의도를 질문하거나 추측할 수 밖에 없는 상황입니다.

그래서 현재 상황에선 빠른 속도를 내는 것에 최적화되어있지만 이후 시점에는 다른 방식의 접근도 필요할 것으로 예상됩니다.

## 디자이너 관련한 문제점 파악

위와 같은 상황과 더불어 현재 제품이 B2B SaaS이고, 제품의 특수성으로 인해 기술적 제약이 대부분의 제약을 차지하고 있습니다. 그래서 기능의 궁극적인 목표(제품 비전이나 전략에서 비롯된)를 이해하고 있더라도 실질적인 구현에 관해서는 기술적 제약사항에 대한 이해가 매우 자주 요구됩니다.

B2C 또는 디자이너가 잘 이해할 수 있는 고객을 대상으로 하는 프로덕트였다면 디자이너가 엔지니어보다 사용자에게 무엇이 어떻게 전달되어야 할지에 대해 전문성을 갖추고 있었을 것입니다.

하지만 지금의 프로덕트에서는 그렇지 못하고 오히려 디자이너 분들이 제품에 대한 이해도가 가장 낮은 상황이어서 디자인 시안 공유가 이루어질 때 백엔드 엔지니어들에 의해 시안이 드랍되는 경우가 굉장히 자주 발생합니다.

Discovery를 위한 와이어프레임이 아니므로 시안에 대한 많은 비판과 그로 인한 재작성은 좋은 일이 아니고, 또한 설령 수정된 시안으로 진행하게 되어도, 시안 자체가 너무 Happy case만을 전제로 하고 있어서 구현 중간에 변경되는 경우도 많고, 그러한 부분 중 일부는 디자인 시안에 반영되지 않아 격차가 발생하게 됩니다.

업무능력이나 경력, 기술적 제약사항 이해도가 백엔드 엔지니어들이 압도적으로 높고, 구현해야 할 기능에 대한 문서가 전혀 없기에, 디자이너는 (때때로 프론트엔드 엔지니어들도) 실제로 시안을 그리거나 구현을 하는 과정에서 백엔드 엔지니어들에게 질문하고 야단인 듯 야단 아닌 야단 같은 피드백을 통해서만 기능에 대한 이해도가 높아지면서 구현 및 배포가 되는 형태로 업무가 진행되고 있습니다.

그러다보니 동료로서 옆에서 보았을 때는 같은 팀 디자이너들이 흔히 말하는 프로덕트 디자이너라기 보다는 UI 디자이너에 머물게 된다고 느껴질 때가 많고, 초기 스타트업 프로덕트 특성상 계속해서 새로운 기능들이 ASAP로 추가되는 경우가 대다수이므로 위와 같은 상황의 개선은 좀처럼 이루어지지 않고 있지만, 마땅한 해결책을 찾지 못하고 있습니다.
